### ISSUE TITLE
[SMV] OSLC Query 3.0 PS01

### ISSUE THREAD

**Issue by berezovskyi** [oasis-open-projects/administration#16]
*Requester name:* Andrew Berezovskyi

*OP name:* Open Services for Lifecycle Collaboration (OSLC) OP

*OP mailing address:* oslc-op@lists.oasis-open-projects.org

*Title:* OSLC Query 3.0 Project Specification 01

*Approval link:* 
https://lists.oasis-open-projects.org/g/oslc-op/topic/intent_to_publish_oslc_query/76785356?p=,,,50,0,0,0::recentpostdate%2Fsticky,,,50,2,0,76785356

*Notes:* 
The PSD (#10) was published without any objections and there are no outstanding tickets, thus we are promoting it to a PS straight away. The publication date is set for Oct 1 and if possible we'd like to request a ballot be opened on Sep 24.

**Comment by OASIS-OP-Admin** [comment]
Since I will be on vacation next week, I set up the ballot now and noted that it will "officially" open on the 24th and close on 10/1. (In other words, if anyone votes before the 24th, that's OK.)   I scheduled a notification email to go to oslc-op@ on 9/23 as a general announcement. The ballot is at https://lists.oasis-open-projects.org/g/oslc-op-pgb/topic/vote_on_oslc_query_version/76871816?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,76871816

**Comment by berezovskyi** [comment]
A reminder to close the ballot, please! Thank you.

**Comment by OASIS-OP-Admin** [comment]
The ballot closed and passed. The result was announced to the PGB and the project mailing list. See https://lists.oasis-open-projects.org/g/oslc-op/topic/oslc_query_version_3_0_is_now/77261711?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,77261711

Approval date is 01 October 2020. 

Staff will now publish the PS and notify the project when it is available.

**Comment by paul-knight** [comment]
It appears that the files are in https://github.com/oslc-op/oslc-specs/releases/download/untagged-5751179e402b612726c6/oslc-query.zip.

This is a single prose file, with an image.

The only issue noted so far is that the links for "This stage" (including the Citation format) have not been updated from psd01 to ps01. The stage name has been properly updated to Project Specification 01 in the subtitle and Citation format.

I will fix this issue.

There do not appear to be any relative links used in this specification, so it should be okay to generate the PDF locally.

**Comment by paul-knight** [comment]
Also noted: hyperlink under "Latest stage" PDF points to HTML instead of PDF.
I'll fix this too.

**Comment by paul-knight** [comment]
Note - this does not include the "description" tag which the other OSLC OP documents have included - I'll add that, based on the Abstract text.
\<meta name="description" content="OSLC Query provides a mechanism for a client to query or search for RDF resources that match a given criteria.">

**Comment by paul-knight** [comment]
During PDF generation, noted the same issues and fixes required during publication of https://github.com/oasis-open-projects/administration/issues/10 (and discussed there):

- following two modifications are made ONLY to a temporary (not published) HTML file used to generate the PDF format.

- Modify image size to avoid running off the page - a simple addition to the \<img> tag/
\<img src="./images/CoreDiscoveryUML.png" width="650">
The original tag was \<img src="./images/CoreDiscoveryUML.png">

- modify the handling of text blocks using \<pre> tags, to avoid truncation on right side, by wrapping the text.
This adds a style definition into the HTML.
pre {
    word-break: break-all;
            overflow-x: auto;
            white-space: pre-wrap;
            white-space: -moz-pre-wrap;
            white-space: -pre-wrap;
            white-space: -o-pre-wrap;
            word-wrap: break-word;
         }

**Comment by paul-knight** [comment]
PDF looks good - entire image is visible, and long lines wrap in the PDF, rather than being truncated.

Note on PDF generation: in running WKHTMLTOPDF to generate the PDF from the HTML, I used the option 
"--disable-smart-shrinking".  If this option is omitted, then the generated PDF uses a nearly-unreadable font size for the entire document. This is because it shrinks the font to allow the longest line of the text blocks to appear on one line. The longest line appears to be located about three lines above the heading 7.7. It's 163 characters long, in a non-proportionally-spaced font.

**Comment by paul-knight** [comment]
Prepared installation scripts, generated installation package.

Requested staging review by OASIS staff, and attaching ZIP file
[query-v3.0-ps01-staging.zip](https://github.com/oasis-open-projects/administration/files/5320379/query-v3.0-ps01-staging.zip)

Please review the staging server installation of the files for OSLC Query v3.0 PS01.

This is a single prose document. Keep an eye out for possible Friday-afternoon slip-ups...

The files are:

OSLC Query Version 3.0
Project Specification 01
01 October 2020

https://docs-staging.oasis-open-projects.org/ztest/oslc-op/query/v3.0/ps01/oslc-query.html
https://docs-staging.oasis-open-projects.org/ztest/oslc-op/query/v3.0/ps01/oslc-query.pdf

Please add comments to https://github.com/oasis-open-projects/administration/issues/16
Also tracked in https://issues.oasis-open.org/browse/TCADMIN-3812.

**Comment by OASIS-OP-Admin** [comment]
That was simple. No problems found. I approve to promote to docs.oasis-open-projects.org.

**Comment by OASIS-OP-Admin** [comment]
Odd. In the HTML, the link for QUERY-39 in the conformance clauses points to the wrong location. It jumps to -24 (the first one listed in Table 7). I spot checked a few others. QUERY-55 jumps to -42. Also happens in the PDF.  Seems random which ones don't link to the correct spot. 

Other than that, the files look OK. I approve to release.

**Comment by paul-knight** [comment]
Since this has no relative links, there was no issue with its PDF files. It should be ready to release with the other three - likely tomorrow, 8 October.

**Comment by paul-knight** [comment]
Removed README warning file - ready for release.

**Comment by paul-knight** [comment]
@chet-ensign Chet will coordinate announcements of the four Project Specifications (Core, CM, RM, Query)

Note the publication URIs are listed in an earlier comment.

**Comment by OASIS-OP-Admin** [comment]
I have announced the publication of the Prj Specs - OASIS homepage (https://www.oasis-open.org/news/announcements/four-new-project-specifications-approved-by-the-oslc-open-project), Twitter, LinkedIn groups, emails (https://lists.oasis-open-projects.org/g/oslc-op/topic/four_new_project/77486324?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,77486324). I also added them to the OASIS Standards page under the Committee/Project Specifications grouping (https://www.oasis-open.org/standards#oslc-query-3.0). I'm checking with Andrew about adding this as a news article to open-services.net. 

No further tasks to do so I'm closing this ticket.
